src/icon/iconview.js

```diff
@@ -22,36 +22,58 @@ export default class IconView extends View {
 	constructor() {
 		super();
 
-		const bind = this.bindTemplate;
+		/**
+		 * The inline SVG or in legacy mode the name of the icon
+		 * which corresponds with the name of the file in the
+		 * {@link module:theme/iconmanagermodel~IconManagerModel}.
+		 *
+		 * @observable
+		 * @member {String} #content
+		 */
+		this.set( 'content' );
 
 		this.template = new Template( {
 			tag: 'svg',
 			ns: 'http://www.w3.org/2000/svg',
 			attributes: {
-				class: [
-					'ck-icon'
-				]
-			},
-			children: [
-				{
-					tag: 'use',
-					ns: 'http://www.w3.org/2000/svg',
-					attributes: {
-						href: {
-							ns: 'http://www.w3.org/1999/xlink',
-							value: bind.to( 'name', name => `#ck-icon-${ name }` )
-						}
-					}
-				}
-			]
+				class: 'ck-icon',
+				viewBox: '0 0 20 20'
+			}
 		} );
+	}
 
+	init() {
 		/**
-		 * The name of the icon. It corresponds with the name of the
-		 * file in the {@link module:theme/iconmanagermodel~IconManagerModel}.
-		 *
-		 * @observable
-		 * @member {String} #name
+		 * This is a hack for lack of innerHTML binding.
+		 * See: https://github.com/ckeditor/ckeditor5-ui/issues/99.
 		 */
+		return super.init().then( () => {
+			this.on( 'change:content', ( evt, name, value ) => {
+				this.element.innerHTML = getIconContent( value );
+			} );
+
+			this.element.innerHTML = getIconContent( this.content );
+		} );
 	}
 }
+
+// Analyzes which type of content is provided and returns it in proper format.
+//
+// Temporarily 2 types of icon content are supported:
+// * Inline SVG - content of svg file as plan text.
+// * Icon name (legacy) - name of icon corresponds with name of the file in the {@link module:theme/iconmanagermodel~IconManagerModel}.
+//
+// @param {String} content
+// @returns {String}
+function getIconContent( content ) {
+	return /</.test( content ) ? extractSvgContent( content ) : `<use xlink:href="#ck-icon-${ content }"></use>`;
+}
+
+// Extracts content which is inside <svg></svg> tag.
+// It removes new line characters and then extracts content of svg.
+//
+// @param {String} svgContent svg xml.
+// @returns {String}
+function extractSvgContent( svgContent ) {
+	return /<[svg][^>]*>(.*)<\/svg>/.exec( svgContent.replace( /\r?\n|\r/g, '' ) )[ 1 ];
```

Comment:
That wasn't the point of the question. We should have one solution. But I'm trying to analyse which of the solutions (currently proposed, spriting, image with data URIs) are ok. E.g. I presume that images with data URIs won't be styleable through CSS, right? Spriting, OTOH, requires a pretty complicated loader for bundlers in use, so this isn't fun too. Hence, the only feasible solution is the one proposed by Oskar. Am I right?